resources/istio/templates/uninstall-job.yaml

```diff
@@ -0,0 +1,44 @@
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  annotations:
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
+    helm.sh/hook: "pre-delete"
+    helm.sh/hook-weight: "10"
+  name: istio-uninstall-job
+spec:
+  activeDeadlineSeconds: 1000
+  backoffLimit: 3
+  template:
+    spec:
+      containers:
+        -
+          command:
+            - bash
+            - "-c"
+            - |
+{{.Files.Get "files/uninstall.sh" | printf "%s" | indent 16}}
+          image: {{ .Values.istio.installer.image }}:{{ .Values.istio.installer.tag }}
+          name: uninstaller
+          volumeMounts:
+          - name: config
+            mountPath: /etc/istio/
+            readOnly: true
+          - name: manifests
+            mountPath: /etc/manifests
+            readOnly: true
+          env:
+          - name: NAMESPACE
+            value: "{{ .Release.Namespace }}"
+          - name: CONFIGMAP_NAME
+            value: kyma-istio-control-plane-config
+      volumes:
+      - name: config
+        configMap:
+          name: kyma-istio-control-plane-config
+      - name: manifests
+        configMap:
+          name: kyma-extra-manifests
+      restartPolicy: Never
+      serviceAccountName: istio-job
```

Comment:
Note: I do not change rbac although SA used for installation does not have delete permissions on couple of resources. Always namespace is delete first so the any 'leftovers' will be cleaned up.